packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -17,30 +17,29 @@ export const preloadMethod = `__vitePreload`
 export const preloadMarker = `__VITE_PRELOAD__`
 
 const preloadHelperId = 'vite/preload-helper'
-const preloadCode = `let scriptRel;const seen = {};export const ${preloadMethod} = ${preload.toString()}`
 const preloadMarkerRE = new RegExp(`"${preloadMarker}"`, 'g')
 
 /**
  * Helper for preloading CSS and direct imports of async chunks in parallel to
  * the async chunk itself.
  */
+
+function detectScriptRel() {
+  // @ts-ignore
+  const relList = document.createElement('link').relList
```

Comment:
Could you try adding `/// <reference lib="dom" />` in the first line of this file and check if it improved the typescript support?
Also you may remove some `// @ts-ignore` if this works.

Reply:
Let's do this in another PR, this change is unrelated to supporting modulepreload